### DISCUSSION TITLE
Is there a way to materialize the table with the pairs for evaluation (and/or get the underlying query)?

### DISCUSSION THREAD

**Discussion by dcaseykc** [moj-analytical-services/splink#2184] [Q&A] [answered]
Is there a way to extract the sql query that represents the (implied?) table of pairs to evaluate for matchy-ness?

From error tracebacks (when I've broken stuff) I know it gets computed as part of a CTE during the prediction step, but my ability to reverse engineer things further is limited.

Mostly, I'm interested in studying the resulting query since splink overall seems cleverly optimized and I'm hoping it'll teach me something.

**Comment by RobinL** [comment, marked as answer]
yes - there are a few options:
```
import logging
logging.getLogger("splink").setLevel(1)
```

If you set that _after_ you instantiate the linker, all SQL will be logged

In addition, each Splink dataframe has a `sql_used_to_create` method and in addition the cache has an `executed_queries` list [self.executed_queries = []](https://github.com/moj-analytical-services/splink/blob/79fa1ac52800de1d5918c6a3925a1df0a5f5f99e/splink/cache_dict_with_logging.py#L13)
